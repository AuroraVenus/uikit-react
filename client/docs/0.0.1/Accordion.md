
## Properties

| Property | Type          | Default | Required | Description |
| -------- |:-------------:| -------:| -------- | ----------- |
| Children | AccordionItem | -       | true     | An array of AccordionItems to be rendered inside the Accordian tag. |
| Options  | string        | -       | true     | A string of component modifiers.For a list of all modifiers see [UIkit Accordion](https://getuikit.com/docs/accordion#component-options). |

## Elements

``` tsx
<Accordion options="">
    <AccordionItem title="String" content="String" />
<Accordion>
```

## Basic Usage

<div>
    <ul uk-tab="">
        <li class="uk-active"><a href="#">Usage</a></li>
        <li><a href="#">Code</a></li>
    </ul>
    <ul class="uk-switcher">
        <li>
            <Accordion width="1-2">
                <AccordionItem 
                    title="Item 1"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 2"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 3" 
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
            </Accordion>
        </li>
        <li>
            <pre>
                <code id="basic-usage" className="language-tsx">
                    &lt;Accordion width=&quot;1-2&quot;&gt;
                    <p>
                        &lt;AccordionItem 
                            title=&quot;Item 1&quot;
                            content=&quot;Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.&quot; /&gt;
                    </p>
                    <p>
                        &lt;AccordionItem 
                            title=&quot;Item 2&quot;
                            content=&quot;Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.&quot; /&gt;
                    </p>
                    <p>
                        &lt;AccordionItem 
                            title=&quot;Item 3&quot; 
                            content=&quot;Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.&quot; /&gt;
                    </p>
                    &lt;/Accordion&gt;
                </code>
            </pre>
        </li>
    </ul>
</div>

## Disable Collapse

``` tsx
<Accordion options="collapsible: false;">
    <AccordionItem 
        title="Item 1"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

    <AccordionItem 
        title="Item 2"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

    <AccordionItem 
        title="Item 3" 
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
</Accordion>
```


## Expand Multiple

``` tsx
<Accordion options="multiple: true;">
    <AccordionItem 
        title="Item 1"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

    <AccordionItem 
        title="Item 2"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

    <AccordionItem 
        title="Item 3" 
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
</Accordion>
```
