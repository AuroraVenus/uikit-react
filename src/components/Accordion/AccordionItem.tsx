import * as React from 'react';
const classNames = require('classnames');
import { setClassNames } from '../../utils/set-class-names'
import { AccordionItemProps } from 'uikit-react';

export class AccordionItem extends React.Component<AccordionItemProps, any> {
    render() {
        return (
            <li>
                <a className="uk-accordion-title" href="#">{this.props.title}</a>
                <div className="uk-accordion-content">{this.props.content}</div>
            </li>
        );
    }
}