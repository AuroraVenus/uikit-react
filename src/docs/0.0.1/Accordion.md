Create a list of items that can be shown individually by clicking an item's header.

-------------
## Properties

| Property | Type          | Default | Required | Description |
| -------- |:-------------:| -------:| -------- | ----------- |
| Children | AccordionItem | -       |    yes   | An array of Items to be rendered inside the Accordian tag. |
| Options  | string        | -       |    yes   | A string of component modifiers.  |

## Modifiers

| Modifier | Value | Default | Description |
| -------- |:-------------:| -------:| -------- |
|`active`| Number |`false`| Index of the element to open initially. |
|`animation`| Boolean |`true`| Reveal item directly or with a transition. |
|`collapsable`| Boolean |`true`| Allow all items to be closed. |
|`content`| String |`> .uk-acordion-content`| The content selector, which selects the accordion content elements. |
|`duration`| Number |`200`| Animation duration in milliseconds. |
|`multiple`| Boolean |`false`| Allow multiple open items. |
|`targets`| String |`> *`| CSS selector of the element(s) to toggle. |
|`toggle`| String |`> .uk-accordion-false`| The toggle selector, which toggles accordion items. |
|`transition`| String |`ease`| The transition to use when revealing items. Use keyword for [easing functions.](https://developer.mozilla.org/en-US/docs/Web/CSS/single-transition-timing-function#Keywords_for_common_timing-functions) |

## Elements

``` tsx
<Accordion options="">
    <AccordionItem title="String" content="String" />
<Accordion>
```

## Basic Usage

<div>
    <ul uk-tab="">
        <li className="uk-active"><a href="#">Usage</a></li>
        <li><a href="#">Code</a></li>
    </ul>
    <ul className="uk-switcher">
        <li>
            <Accordion width="1-2">
                <AccordionItem 
                    title="Item 1"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 2"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 3" 
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
            </Accordion>
        </li>
        <li>
            <pre>
                <Code code='
                    <Accordion>
                        <AccordionItem 
                            title="Item 1"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 2"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 3" 
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
                    </Accordion>'
                />
            </pre>
        </li>
    </ul>
</div>

## Disable Collapse

<div>
    <ul uk-tab="">
        <li className="uk-active"><a href="#">Usage</a></li>
        <li><a href="#">Code</a></li>
    </ul>
    <ul className="uk-switcher">
        <li>
            <Accordion options="collapsible: false;">
                <AccordionItem 
                    title="Item 1"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 2"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 3" 
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
            </Accordion>
        </li>
        <li>
            <pre>
                <Code code='
                    <Accordion options="collapsible: false;">
                        <AccordionItem 
                            title="Item 1"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 2"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 3" 
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
                    </Accordion>'
                />
            </pre>
        </li>
    </ul>
</div>


## Expand Multiple

<div>
    <ul uk-tab="">
        <li className="uk-active"><a href="#">Usage</a></li>
        <li><a href="#">Code</a></li>
    </ul>
    <ul className="uk-switcher">
        <li>
            <Accordion options="collapsible: false;">
                <AccordionItem 
                    title="Item 1"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 2"
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                <AccordionItem 
                    title="Item 3" 
                    content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
            </Accordion>
        </li>
        <li>
            <pre>
                <Code code='
                    <Accordion options="multiple: true;">
                        <AccordionItem 
                            title="Item 1"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 2"
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />

                        <AccordionItem 
                            title="Item 3" 
                            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
                    </Accordion>'
                />
            </pre>
        </li>
    </ul>
</div>
