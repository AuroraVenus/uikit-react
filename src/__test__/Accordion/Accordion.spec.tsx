import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Accordion, AccordionItem } from './../../components/Accordion'

describe('<Accordion>', () => {
  it('Should render with default props', () => {
    const wrapper = renderer.create(
      <Accordion>
        {' '}
        <AccordionItem title="Heading" content="Hello" />
      </Accordion>,
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render with collapsible option FALSE', () => {
    const options = {
      collapsible: false,
    }
    const wrapper = renderer.create(
      <Accordion options={options}>
        <AccordionItem title="Heading" content="Hello" />
      </Accordion>,
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should handle events ', () => {
    const props = {
      onBeforeHide: jest.fn(),
      onBeforeShow: jest.fn(),
    }
    const wrapper = shallow(
      <Accordion {...props}>
        <AccordionItem title="Heading" content="Hello" />
      </Accordion>,
    )
  })
})
