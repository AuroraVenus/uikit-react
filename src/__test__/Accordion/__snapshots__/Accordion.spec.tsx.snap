// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion> Should render with collapsible option FALSE 1`] = `
<ul
  className=""
  id={null}
  style={null}
>
  <li>
    <a
      className="uk-accordion-title"
      role="button"
    >
      Heading
    </a>
    <div
      className="uk-accordion-content"
    >
      Hello
    </div>
  </li>
</ul>
`;

exports[`<Accordion> Should render with default props 1`] = `
<ul
  className=""
  id={null}
  style={null}
>
   
  <li>
    <a
      className="uk-accordion-title"
      role="button"
    >
      Heading
    </a>
    <div
      className="uk-accordion-content"
    >
      Hello
    </div>
  </li>
</ul>
`;
